flexmeasures/data/models/planning/shiftable_load.py

```diff
@@ -90,38 +93,25 @@ def compute(self) -> pd.Series | None:
             name="event_value",
         )
 
-        # optimize schedule for tomorrow. We can fill len(schedule) rows, at most
-        rows_to_fill = min(
-            ceil(duration / consumption_price_sensor.event_resolution), len(schedule)
-        )
-
         # convert power to energy using the resolution of the sensor.
         # e.g. resolution=15min, power=1kW -> energy=250W
         energy = power * consumption_price_sensor.event_resolution / timedelta(hours=1)
 
-        if rows_to_fill > len(schedule):
-            raise ValueError(
-                f"Duration of the period exceeds the schedule window. The resulting schedule will be trimmed to fit the planning window ({start}, {end})."
-            )
+        # we can fill duration/resolution rows or, if the duration is larger than the schedule
+        # window, fill the entire window.
+        rows_to_fill = min(
+            ceil(duration / consumption_price_sensor.event_resolution), len(schedule)
+        )
 
-        # check if the time_restrictions allow for a load of the duration provided
-        if load_type in [LoadType.INFLEXIBLE, LoadType.SHIFTABLE]:
-            # get start time instants that are not feasible, i.e. some time during the ON period goes through
-            # a time restriction interval
-            time_restrictions = (
-                time_restrictions.rolling(duration).max().shift(-rows_to_fill + 1)
-            )
-            time_restrictions = (time_restrictions == 1) | time_restrictions.isna()
+        if rows_to_fill == len(schedule):
+            schedule[:] = energy
```

Comment:
Aren't we forgetting about `time_restrictions` here? Or are we ignoring them?

Reply:
Good catch!

If `len(time_restrictions) > 0` and `duration > len(schedule)` we have several options:
1. raise an Exception
2. schedule the load for the window duration
3. return an empty schedule

Reply:
I guess time_restricions should not be ignored. An empty schedule is misleading and it's hard to detect that something is not done right. 

Do we not have an official way to handle infeasible problems when we run storage scheduling?

Reply:
In case of an infeasible problem, the StorageScheduler returns a fallback schedule. On the other hand, the input data is used to check if it could generate a problem with feasible constraints, and raises a ValueError exception, otherwise.

Reply:
This is the only remaining point I think @nhoening

Reply:
Okay, let's raise (if there are also time_restrictions)